specification/monitor/resource-manager/Microsoft.Insights/stable/2023-10-01/metrics_API.json

```diff
@@ -0,0 +1,610 @@
+          "type": "string"
+        },
+        "top": {
+          "description": "The maximum number of records to retrieve.\nValid only if $filter is specified.\nDefaults to 10.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "orderBy": {
+          "description": "The aggregation to use for sorting results and the direction of the sort.\nOnly one order can be specified.\nExamples: sum asc.",
+          "type": "string"
+        },
+        "rollUpBy": {
+          "description": "Dimension name(s) to rollup results by. For example if you only want to see metric values with a filter like 'City eq Seattle or City eq Tacoma' but don't want to see separate values for each city, you can specify 'RollUpBy=City' to see the results for Seattle and Tacoma rolled up into one timeseries.",
+          "type": "string"
+        },
+        "resultType": {
+          "description": "Reduces the set of data collected. The syntax allowed depends on the operation. See the operation's description for details.",
+          "type": "string",
+          "enum": [
+            "Data",
+            "Metadata"
+          ],
+          "x-ms-enum": {
+            "name": "MetricResultType",
+            "modelAsString": true
+          }
+        },
+        "metricNamespace": {
+          "description": "Metric namespace where the metrics you want reside.",
+          "type": "string"
+        },
+        "autoAdjustTimegrain": {
+          "description": "When set to true, if the timespan passed in is not supported by this metric, the API will return the result using the closest supported timespan. When set to false, an error is returned for invalid timespan parameters. Defaults to false.",
+          "type": "boolean"
+        },
+        "validateDimensions": {
+          "description": "When set to false, invalid filter parameter values will be ignored. When set to true, an error is returned for invalid filter parameters. Defaults to true.",
+          "type": "boolean"
+        }
+      }
+    },
+    "Unit": {
+      "type": "string",
+      "description": "The unit of the metric.",
+      "enum": [
+        "Count",
+        "Bytes",
+        "Seconds",
+        "CountPerSecond",
+        "BytesPerSecond",
+        "Percent",
+        "MilliSeconds",
+        "ByteSeconds",
+        "Unspecified",
+        "Cores",
+        "MilliCores",
+        "NanoCores",
+        "BitsPerSecond"
+      ],
+      "x-ms-enum": {
+        "name": "Unit",
```

Comment:
Old swagger had the name as `MetricUnit`. Should we link this instead to the MetricUnit defined in the common monitoring types file? It's also defined there. Currently, the code generator is generating both a 'Unit' and a 'MetricUnit' model.